src/interfaces.rs

```diff
@@ -91,6 +93,12 @@ impl Interfaces {
         self.interfaces.remove(interface_name)
     }
 
+    /// Remove the interfaces which name is in the HashSet
+    pub(crate) fn remove_many(&mut self, to_remove: &HashSet<&str>) {
+        self.interfaces
+            .retain(|k, _v| !to_remove.contains(k.as_str()));
+    }
```

Comment:
This is just to give and idea, we should do something like this for the following reasons:

- do not allocate an additional set
- the to_remove length set will always be less than the interfaces

```suggestion
    pub(crate) fn remove_many<I, S>(&mut self, to_remove: I) where I: IntoIter<Item = S>, S: AsRef<str> {
        for i in to_remove {
           self.remove(i.as_ref());
        }
    }
```